ale_linters/sml/smlnj.vim

```diff
@@ -12,20 +12,34 @@ function! ale_linters#sml#smlnj#Handle(buffer, lines) abort
 
     let l:out = []
     let l:pattern = '^.*\:\([0-9\.]\+\)\ \(\w\+\)\:\ \(.*\)'
+    let l:pattern2 = '^.*\:\([0-9]\+\)\.\?\([0-9]\+\).* \(\(Warning\|Error\): .*\)'
 
     for l:line in a:lines
+        let l:match2 = matchlist(l:line, l:pattern2)
+
+        if len(l:match2) != 0
+          call add(l:out, {
+          \   'bufnr': a:buffer,
+          \   'lnum': l:match2[1] + 0,
+          \   'col' : l:match2[2] + 0,
+          \   'text': l:match2[3],
+          \   'type': l:match2[3] =~ '^Warning' ? 'W' : 'E',
```

Comment:
There's a Vint issue here. Look at Travis CI. You'll need to use `=~#` or `=~?`. It depends on what you want.